handler.go

```diff
@@ -20,6 +20,20 @@ import (
 	"github.com/filecoin-project/go-jsonrpc/metrics"
 )
 
+type RawParams json.RawMessage
+
+var rtRawParams = reflect.TypeOf(RawParams{})
+
+// todo is there a better way to tell 'struct with any number of fields'?
+func DecodeParams[T any](p RawParams) (T, error) {
+	var t T
+	err := json.Unmarshal(p, &t)
+
+	// todo also handle list-encoding automagically (json.Unmarshal doesn't do that, does it?)
```

Comment:
Do we need this now?

Reply:
Nope, it's a nice to have, but not needed afaict. The spec doesn't say that endpoints need to support both, so it's up to the endpoint really.